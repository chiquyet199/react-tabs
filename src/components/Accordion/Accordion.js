import React, {Component} from "react";
import {TouchableOpacity, View} from "react-native";
import {bool, func} from "prop-types";
import Collapsible from "react-native-collapsible";
import {isWeb} from "../../utils/common";

export const CompatibleCollapsible = props => {
  return isWeb ? (
    <View>{props.collapsed ? null : props.children}</View>
  ) : (
    <Collapsible {...props}>{props.children}</Collapsible>
  );
};
CompatibleCollapsible.propTypes = {
  collapsed: bool,
};

export class Accordion extends Component {
  state = {
    collapsed: true,
  };

  handlePress = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  render() {
    const {title: Title} = this.props;

    return (
      <TouchableOpacity onPress={this.handlePress}>
        <Title collapsed={this.state.collapsed} />
        <CompatibleCollapsible collapsed={this.state.collapsed}>
          {this.props.children}
        </CompatibleCollapsible>
      </TouchableOpacity>
    );
  }
}

Accordion.propTypes = {
  title: func.isRequired,
};

export default Accordion;
