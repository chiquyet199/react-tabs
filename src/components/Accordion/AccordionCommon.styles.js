import React from "react";
import Icon from "../Icon";
import {themed} from "../../utils/common";
import styledNative from "styled-components/native";
import {bool} from "prop-types";

export const ThemedText = themed(styledNative.Text`
  color: ${({globaltheme, theme}) => theme[globaltheme].textColor};
`);
export const TitleRow = styledNative.View`
  flex-direction: row;
  align-items: center;
`;
export const ThemedIcon = themed(props => (
  <Icon {...props} color={props.theme[props.globaltheme].textColor} />
));
export const AccordionOpenIndicator = ({collapsed}) => (
  <ThemedIcon name={collapsed ? "chevron-down" : "chevron-up"} size={32} />
);
AccordionOpenIndicator.propTypes = {
  collapsed: bool,
};
