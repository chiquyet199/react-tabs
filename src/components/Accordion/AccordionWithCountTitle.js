import React from "react";
import {View} from "react-native";
import {bool, number, string} from "prop-types";
import {Title, CountWrapper, Count} from "./AccordionWithCountTitle.styles";
import {
  AccordionOpenIndicator,
  TitleRow,
  ThemedIcon,
} from "./AccordionCommon.styles";

export const AccordionWithCountTitle = ({
  children,
  headerCount,
  headerIconName,
  collapsed,
  title,
}) => (
  <View>
    <TitleRow>
      <ThemedIcon name={headerIconName} size={32} />
      <Title>{title}</Title>
      <CountWrapper>
        <Count>{headerCount}</Count>
      </CountWrapper>
      <AccordionOpenIndicator collapsed={collapsed} />
    </TitleRow>
    <View>{children}</View>
  </View>
);
AccordionWithCountTitle.propTypes = {
  headerIconName: string.isRequired,
  headerCount: number.isRequired,
  collapsed: bool,
  title: string.isRequired,
};

export default AccordionWithCountTitle;
