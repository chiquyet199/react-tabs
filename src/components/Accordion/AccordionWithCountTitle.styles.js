import {themed} from "../../utils/common";
import styledNative from "styled-components/native";
import {ThemedText} from "./AccordionCommon.styles";

export const Title = styledNative(ThemedText)`
  flex-grow: 2;
  font-weight: bold;
  font-size: 18;
`;
export const TitleRow = styledNative.View`
  flex-direction: row;
  align-items: center;
`;
export const CountWrapper = themed(styledNative.Text`
  background-color: ${({globaltheme, theme}) => theme[globaltheme].boxBorder};
  border-radius: 12;
  height: 26;
  overflow: hidden;
  text-align: center;
  width: 26;
`);
export const Count = styledNative(ThemedText)`
  line-height: 26;
  text-align: center;
`;
