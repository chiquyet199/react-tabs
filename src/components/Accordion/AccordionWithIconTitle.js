import React from "react";
import {View} from "react-native";
import {bool, string} from "prop-types";
import styledNative from "styled-components/native";
import {
  AccordionOpenIndicator,
  TitleRow,
  ThemedIcon,
  ThemedText,
} from "./AccordionCommon.styles";

const Title = styledNative(ThemedText)`
  font-size: 24;
  line-height: 30;
`;
const SubTitle = styledNative(ThemedText)`
  font-size: 14;
  line-height: 18;
`;
const TitleTextWrapper = styledNative.View`
  flex-grow: 1;
`;

export const AccordionWithIconTitle = ({
  children,
  headerIconName,
  collapsed,
  subTitle,
  title,
}) => (
  <View>
    <TitleRow>
      <ThemedIcon name={headerIconName} size={32} />
      <TitleTextWrapper>
        <Title>{title}</Title>
        <SubTitle>{subTitle}</SubTitle>
      </TitleTextWrapper>
      <AccordionOpenIndicator collapsed={collapsed} />
    </TitleRow>
    <View>{children}</View>
  </View>
);
AccordionWithIconTitle.propTypes = {
  headerIconName: string.isRequired,
  collapsed: bool,
  subTitle: string.isRequired,
  title: string.isRequired,
};

export default AccordionWithIconTitle;
