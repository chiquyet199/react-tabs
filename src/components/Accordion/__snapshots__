/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Accordion/__native_snapshots__