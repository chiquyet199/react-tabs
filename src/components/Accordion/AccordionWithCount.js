import React from "react";
import {number, string} from "prop-types";
import AccordionWithCountTitle from "./AccordionWithCountTitle";
import {Accordion} from "./Accordion";

export const AccordionWithCount = props => (
  <Accordion
    title={({collapsible}) => (
      <AccordionWithCountTitle
        {...{
          ...props,
          collapsible,
        }}
      />
    )}
  >
    {props.children}
  </Accordion>
);

AccordionWithCount.propTypes = {
  headerIconName: string.isRequired,
  headerCount: number.isRequired,
  title: string.isRequired,
};

export default AccordionWithCount;
