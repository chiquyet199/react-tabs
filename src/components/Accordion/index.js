import Accordion from "./Accordion";
import AccordionWithCount from "./AccordionWithCount";
import AccordionWithIconTitle from "./AccordionWithIconTitle";

export {Accordion, AccordionWithCount, AccordionWithIconTitle};
