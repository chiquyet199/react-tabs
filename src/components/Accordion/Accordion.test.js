import React from "react";
import {Platform, Text, TouchableOpacity, View} from "react-native";
import {shallow} from "enzyme";
import Collapsible from "react-native-collapsible";
import {Accordion, CompatibleCollapsible} from "./Accordion";
import {isWeb} from "../../utils/common";

describe("Accordion test", () => {
  it(`matches snapshot on ${Platform.OS}`, () => {
    expect(getComponent()).toMatchSnapshot();
  });

  it("passes collapsed as default to title", () => {
    const component = getComponent();
    expect(component.find(SampleTitle)).toHaveProp("collapsed", true);
  });

  it("changes collapsed to false on title after click on TouchableOpacity", () => {
    const component = getComponent();
    component.find(TouchableOpacity).simulate("press");
    expect(component.find(SampleTitle)).toHaveProp("collapsed", false);
  });

  it("passes collapsed as default to CompatibleCollapsible", () => {
    const component = getComponent();
    expect(component.find(CompatibleCollapsible)).toHaveProp("collapsed", true);
  });

  it("changes collapsed to false on CompatibleCollapsible after click on TouchableOpacity", () => {
    const component = getComponent();
    component.find(TouchableOpacity).simulate("press");
    expect(component.find(CompatibleCollapsible)).toHaveProp(
      "collapsed",
      false,
    );
  });

  describe("CompatibleCollapsible tests", () => {
    it(`CompatibleCollapsible matches snapshot on ${Platform.OS}`, () => {
      expect(getCompatibleCollapsible()).toMatchSnapshot();
    });

    it("doesn't render children on web / passes collapsed as true on native when collapsed is true", () => {
      const collapsed = true;
      const component = getCompatibleCollapsible(collapsed);
      if (isWeb) {
        expect(component.find(View).children()).not.toExist();
      } else {
        expect(component.find(Collapsible)).toHaveProp("collapsed", collapsed);
      }
    });

    const getCompatibleCollapsible = (collapsed = false) => {
      return shallow(
        <CompatibleCollapsible collapsed={collapsed}>
          <View>
            <Text>Lorem ipsum dolor sic amet</Text>
          </View>
        </CompatibleCollapsible>,
      );
    };
  });

  const getComponent = (
    props,
    children = (
      <View>
        <Text>Lorem ipsum</Text>
      </View>
    ),
  ) => {
    const parsedProps = {
      title: SampleTitle,
      props,
    };

    return shallow(<Accordion {...parsedProps}>{children}</Accordion>);
  };

  const SampleTitle = ({collapsed}) => (
    <View style={{opacity: collapsed ? 0.5 : 1}}>
      <Text>Sample title</Text>
    </View>
  );
});
