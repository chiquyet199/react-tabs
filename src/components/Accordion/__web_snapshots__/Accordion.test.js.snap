// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion test CompatibleCollapsible tests CompatibleCollapsible matches snapshot on web 1`] = `
<View>
  <View>
    <Text>
      Lorem ipsum dolor sic amet
    </Text>
  </View>
</View>
`;

exports[`Accordion test matches snapshot on web 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
>
  <SampleTitle
    collapsed={true}
  />
  <CompatibleCollapsible
    collapsed={true}
  >
    <View>
      <Text>
        Lorem ipsum
      </Text>
    </View>
  </CompatibleCollapsible>
</TouchableOpacity>
`;
